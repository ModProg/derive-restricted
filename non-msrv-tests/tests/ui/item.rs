use std::marker::PhantomData;

use derive_where::derive_where;

#[derive_where]
struct NoOption<T>(PhantomData<T>);

#[derive_where()]
struct EmptyAttribute<T>(PhantomData<T>);

#[derive_where(Clone; T;)]
struct SemiColonAtTheEnd<T, U>(T, PhantomData<U>);

#[derive_where(Clone; T,,)]
struct DoubleColonAtTheEnd<T, U>(T, PhantomData<U>);

#[derive_where(Clone; where)]
struct InvalidGeneric<T>(PhantomData<T>);

#[derive_where(Clone Debug)]
struct MissingCommaBetweenTraits<T>(PhantomData<T>);

#[derive_where(Clone; T U)]
struct MissingCommaBetweenGenerics<T, U, V>(T, PhantomData<(U, V)>);

#[derive_where("Clone")]
struct InvalidTrait<T>(PhantomData<T>);

fn main() {}
