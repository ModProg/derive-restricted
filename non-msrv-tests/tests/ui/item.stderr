error: empty `derive_where` found
 --> tests/ui/item.rs:5:1
  |
5 | #[derive_where]
  | ^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `derive_where` (in Nightly builds, run with -Z macro-backtrace for more info)

error: empty `derive_where` found
 --> tests/ui/item.rs:8:1
  |
8 | #[derive_where()]
  | ^^^^^^^^^^^^^^^^^
  |
  = note: this error originates in the attribute macro `derive_where` (in Nightly builds, run with -Z macro-backtrace for more info)

error: expected `,`
  --> tests/ui/item.rs:11:24
   |
11 | #[derive_where(Clone; T;)]
   |                        ^

error: expected type to bind to, expected one of: `for`, parentheses, `fn`, `unsafe`, `extern`, identifier, `::`, `<`, square brackets, `*`, `&`, `!`, `impl`, `_`, lifetime
  --> tests/ui/item.rs:14:25
   |
14 | #[derive_where(Clone; T,,)]
   |                         ^

error: expected type to bind to, expected one of: `for`, parentheses, `fn`, `unsafe`, `extern`, identifier, `::`, `<`, square brackets, `*`, `&`, `!`, `impl`, `_`, lifetime
  --> tests/ui/item.rs:17:23
   |
17 | #[derive_where(Clone; where)]
   |                       ^^^^^

error: expected `;` or `,
  --> tests/ui/item.rs:20:22
   |
20 | #[derive_where(Clone Debug)]
   |                      ^^^^^

error: expected `,`
  --> tests/ui/item.rs:23:25
   |
23 | #[derive_where(Clone; T U)]
   |                         ^

error: unexpected option syntax
  --> tests/ui/item.rs:26:16
   |
26 | #[derive_where("Clone")]
   |                ^^^^^^^

warning: unused import: `std::marker::PhantomData`
 --> tests/ui/item.rs:1:5
  |
1 | use std::marker::PhantomData;
  |     ^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = note: `#[warn(unused_imports)]` on by default
